plugin/trino-ldap-group-provider/src/test/java/io/trino/plugin/ldapgroup/MockLdapClient.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ldapgroup;
+
+import io.trino.plugin.base.ldap.LdapClient;
+import io.trino.plugin.base.ldap.LdapQuery;
+
+import javax.naming.NamingEnumeration;
+import javax.naming.NamingException;
+import javax.naming.directory.SearchResult;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MockLdapClient
```

Comment:
Can we avoid MockLdapClient ? Instead we could use a LDAP server via test containers to perform additional assertion.

Reply:
We have integration tests as well that use an LDAP container, see `plugin/trino-ldap-group-provider/src/test/java/io/trino/plugin/ldapgroup/TestLdapGroupProviderIntegration.java`
Mocking is used to test error handing and behaviour that can't easly be achieved with an actual LDAP server (e.g., malformed data, exceptions)

Reply:
Can we rename it as `TesingLdapClient` ?